iocage_lib/ioc_json.py

```diff
@@ -580,12 +580,33 @@ def get_mac_prefix():
             # Helps avoid clashes with other systems in the network
             mac_prefix = default_mac[0]['addr'].replace(':', '')[:6]
 
-            return mac_prefix
         except KeyError:
             # They don't have a default gateway, opting for generation of mac
             mac = random.randint(0x00, 0xfffff)
 
-            return f'{mac:06x}'
+            mac_prefix = f'{mac:06x}'
+
+        # Reason for this change is that the first bit in the first byte of
+        # mac address dictates unicast/multicast address. In case of
+        # multicast address, bridge does not learn from such addresses.
+        # So we make sure that we have it unset and the second bit indicates
+        # that this mac is being used in a local network which we set it
+        # always.
```

Comment:
The only thing I would add here is a note about network endian-ness. That networks are MSB ordered. It speaks to the fact that the mentioned first- and second-bit are at the far-right of the byte, driving-home exactly why indices 6 and 7 are checked from the [zero-indexed] list representing each bit